src/charm.py

```diff
@@ -153,25 +213,52 @@ def _on_collect_status(self, event: CollectStatusEvent):
         if not self.tls.tls_file_pushed_to_workload(File.CA):
             event.add_status(WaitingStatus("Waiting for CA certificate in workload"))
             return
+        if not self.tls.tls_file_available_in_charm(File.CA):
+            event.add_status(WaitingStatus("Certificate is unavailable in the charm"))
+            return
         if not self._is_vault_service_started():
             event.add_status(WaitingStatus("Waiting for Vault service to start"))
             return
+        self._collect_vault_app_status(event)
```

Comment:
This feels strange. Is this the recommended approach? Why don't we observe `collect_app_status` instead?

Reply:
This was just a quick way to get ruff to stop complaining about complexity and get me moving again. You're right I should revisit this.